docs/install/build_from_source.md

```diff
@@ -317,6 +317,40 @@ Use [CMake](https://cmake.org/) to create a Visual Studio solution in ```./build
 In Visual Studio, open the solution file,```.sln```, and compile it.
 These commands produce a library called ```mxnet.dll``` in the ```./build/Release/``` or ```./build/Debug``` folder.
 
+</div>
+
+<div class="linux ubuntu">
+
+## Build MXNet using NCCL
+- Download and install the latest NCCL library from NVIDIA.
+- Note the directory path in which NCCL libraries and header files are installed.
+- Ensure that the installation directory contains ```lib``` and ```include``` folders.
+- Ensure that the prerequisites for using NCCL such as Cuda libraries are met. 
+- Append the ```config.mk``` file with following, in addition to the CUDA related options.
+- USE_NCCL=1
+- USE_NCCL_PATH=path-to-nccl-installation-folder
+``` bash
+echo "USE_NCCL=1" >> make/config.mk
+echo "USE_NCCP_PATH=path-to-nccl-installation-folder" >> make/config.mk
+cp make/config.mk .
+```
+- Run make command
+``` bash
+make -j"$(nproc)"
+```
+
+## Validation
+- Follow the steps to install MXNet Python binding.
+- Comment the following line in ```test_nccl.py``` file at ```incubator-mxnet/tests/python/gpu/test_nccl.py```
+``` bash
+@unittest.skip("Test requires NCCL library installed and enabled during build")
+```
+- Run test_nccl.py script as follows. The test should complete. It does not produce any output.
+
+## Recommendation for best performance
+It is recommended to set environment variable NCCL_LAUNCH_MODE to PARALLEL when using NCCL version 2.1 or newer.
```

Comment:
Shall we add this env var `NCCL_LAUNCH_MODE ` to https://github.com/apache/incubator-mxnet/blob/master/docs/faq/env_var.md as well? What are the available options and what is the difference? @ptrendx

Reply:
This env var is not MXNet specific - it is NCCL's env var that tells NCCL whether to use cooperative launch feature of CUDA 9. The possible options are PARALLEL (older one, which currently is faster) and GROUP (cooperative launch, currently slower).

Reply:
Ideally mxnet users should be able to get env var info in that page without searching for nccl page. Consider adding a section for nccl in that page